Tools/TestWebKitAPI/Tests/WebKitCocoa/SessionStorage.mm

```diff
@@ -147,3 +148,179 @@ static void runSessionStorageInNewWindowTest(NSString* createWindowJS, ShouldSes
     }];
     TestWebKitAPI::Util::run(&readyToContinue);
 }
+
+namespace TestWebKitAPI {
+
+static void synchronouslyLoadUrl(RetainPtr<TestWKWebView>& webView, RetainPtr<TestNavigationDelegate>& delegate, NSString* url)
```

Comment:
Is there a reason an API test is preferred than a layout test for this change?
(there's cross-origin test like: LayoutTests/http/tests/security/cross-origin-indexeddb.html)

Reply:
No, mostly due to still learning, but my understanding is that an API test allows for testing across many origins, whereas layout tests are limited to localhost and loopback address. The API test currently relies on four domains - I can try adding a layout test for this, as well, if you prefer that.

Reply:
I think we usually use API test for an API change or something that is untestable with layout tests (e.g. process crash and relaunch). More importantly, tests in SessionStorage.mm only runs on Cocoa platform, and this is not a platform specific change. (I am not sure why we need four domains for testing; it seems {a, b, a/b, b/a} is enough).

Reply:
That makes sense. This change doesn't need four domains, but I wanted to be sure I didn't miss an edge case. I did find a bug in an earlier version of this patch because I tested both http and https for the same domain. I'll add a layout test.

Reply:
With the addition of the layout test, should I delete the API test? I believe the API test provides slightly better coverage with both cross-site and cross-origin tests, but the new layout test should be sufficient for the general case.